src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -1194,12 +1194,20 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		let validInputs = this.validateInputs();
 		if (validInputs) {
 			if (this.useConnectionString) {
-				const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
-				if (connInfo) {
+				try {
+					const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
+					if (!connInfo) {
+						this._logService.error(`${this._providerName}.buildConnectionInfo returned an undefined value.`)
+						this._errorMessageService.showDialog(Severity.Error, localize('connectionWidget.Error', "Error"), localize('connectionWidget.ConnectionStringError', "Failed to parse the connection string."),);
+						return false;
+					}
 					model.options = connInfo.options;
 					model.savePassword = true;
-				} else {
-					this._errorMessageService.showDialog(Severity.Error, localize('connectionWidget.Error', "Error"), localize('connectionWidget.ConnectionStringError', "Failed to parse the connection string."));
+				} catch (err) {
+					this._logService.error(`Failed to parse the connection string : ${err}`)
+					this._errorMessageService.showDialog(Severity.Error, localize('connectionWidget.Error', "Error"),
+						localize('connectionWidget.ConnectionStringError', "Failed to parse the connection string."),
```

Comment:
And for the message I would say it should be `Failed to parse the connection string. {0}` where 0 is the message (you can use getErrorMessage from vs/base/common/errors for that).

Then the detail information would just be err.stack

Reply:
you mean the messageDetails? It won't be good for the message itself since the exception message is not localized and would be visible right in the alert box.

Reply:
No, the message. It's fine to have some bits of non-localized text if a provider isn't localizing their error messages as long as the parts we control are localized and we do a best effort here. It's expected that extensions should be localizing any text that might be displayed to the user. And in cases that you can't fully (such as yours) then that's acceptable enough - although you are free to take the original error and throw your own error with localized text if you want.

We don't want people to have to dig into the message details just to find out what actually happened though if we can avoid it.

Reply:
ok, cool, this makes sense.

Reply:
Ok this is how it looks now with a935073:

<img width="647" alt="grafik" src="https://user-images.githubusercontent.com/951587/226757934-e15412a6-aec1-4f21-becd-94be2aef865c.png">